### PR TITLE
Refactor #230 implementation

### PR DESCRIPTION
**Pull request by seratch**
This pull request improves the changes in #230 

@jwszolek Can you check the changes and verify if it works with your real AWS data?

### CODE REVIEW

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 1]
```diff
@@ -24,10 +27,6 @@ object Table {
 
 class hashPK extends StaticAnnotation
 class rangePK extends StaticAnnotation
-case class AnnotationMeta(
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:27]: we don't want to make this public

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 2]
```diff
@@ -24,10 +27,6 @@ object Table {
...
-  name: String,
-  annotation: String,
-  typeSignature: Any)
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:30]: typeSignature is actually unused

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 3]
```diff
@@ -70,80 +69,68 @@ case class Table(
...
-              case "String" => f.asInstanceOf[String]
-            }
-          })
+    val constructorArgs: Seq[AnnotatedConstructorArgMeta] = extractAnnotatedConstructorArgs(entity)
+    val getterCallResults: Seq[(String, AnyRef)] = extractGetterNameAndValue(entity)
+
+    var maybeHashPK: Option[Any] = None
+    var maybeRangePK: Option[Any] = None
+    val attributes: ListBuffer[(String, AnyRef)] = ListBuffer()
+    for (arg <- constructorArgs) {
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:84]: reduce loops

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 4]
```diff
@@ -70,80 +69,68 @@ case class Table(
...
-    val attrs = getterNamesFromEntity(entity)
-    dynamoDB.put(this, hashPK, rangePK, attrs: _*)
-  }
+  private case class AnnotatedConstructorArgMeta(name: String, annotationNames: Seq[String])
 
-  private def getterAnnotationsFromEntity[T: u.TypeTag](entity: T): List[AnnotationMeta] = {
+  private def extractAnnotatedConstructorArgs[T: u.TypeTag](entity: T): Seq[AnnotatedConstructorArgMeta] = {
     u.typeOf[entity.type].decl(termNames.CONSTRUCTOR).asMethod.paramLists.flatten
       .collect({
-        case t: TermSymbol if (t.isVal && t.annotations.nonEmpty) =>
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:125]: it's okay to have `var` here - thus, I removed the `val` check

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 5]
```diff
@@ -70,80 +69,68 @@ case class Table(
...
-          AnnotationMeta(t.name.toString, t.annotations.head.toString, t.typeSignature.toString)
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:126]: `t.annotations.head.toString` does not work if the field have a different annotation as the first one.

File: dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala [hunk 6]
```diff
@@ -70,80 +69,68 @@ case class Table(
...
-  }
-
-  private def getterNamesFromEntity(obj: Any): Seq[(String, AnyRef)] = {
-    val fieldNames = obj.getClass.getDeclaredFields
-      .filter(f => Modifier.isPrivate(f.getModifiers))
-      .filterNot(f => Modifier.isStatic(f.getModifiers))
+        case t if t != null && t.annotations.nonEmpty =>
+          Some(AnnotatedConstructorArgMeta(
+            name = t.name.toString,
+            // FIXME: should we use canonical name?
```
Comment (seratch) [inline dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala:110]: including package name may be safer

**Review by seratch — COMMENTED** [review]
comments for reviewers

**Review by jwszolek — APPROVED** [review]
hey @seratch, thanks for all improvements added!

**Comment by seratch** [comment]
@jwszolek cool - thanks. I will release a new version.

**Pull request merged**